interpret.scm

```diff
@@ -89,7 +130,7 @@
   (lambda (parse_tree state return break continue)
     (cond
       ((null? parse_tree) state)
-      (else (interpret_parse_tree (parse_tree_remainder parse_tree) (Mstate-cps (parse_tree_statement parse_tree) state return break continue) return break continue))
+      (else (interpret_parse_tree (parse_tree_remainder parse_tree) (Mstate_outer (parse_tree_statement parse_tree) state return) return break continue))
```

Comment:
I think you need Mstate-cps here, not Mstate_outer because of function definitions inside of other functions. When you call the state_remainder function on the state it will give you everything in your state before the function definition (which could include non-global variables that you need in your nested definition).